File: Data Structures & Algorithms/invert-a-binary-tree/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public TreeNode InvertTree(TreeNode root) {
        Invert(root);
        return root;
    }

    public void Invert(TreeNode node) {
        if (node is null) {
            return;
        }

        (node.left, node.right) = (node.right, node.left);

        Invert(node.left);
        Invert(node.right);
    }
}

File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        nodes = []
        def preorder(node):
            if not node:
                return
            
            preorder(node.right)
            preorder(node.left)
            nodes.append(node.val)
        
        preorder(root)
        nodes.reverse()
        return nodes